src/tof/chopper.py

```diff
@@ -75,3 +79,58 @@ def __repr__(self) -> str:
             f"frequency={self.frequency:c}, phase={self.phase:c}, "
             f"cutouts={len(self.open)})"
         )
+
+    def as_readonly(self):
+        return ReadonlyChopper(self)
+
+
+class ReadonlyChopper(Component):
+    def __init__(self, chopper: Chopper):
+        self._frequency = chopper.frequency
+        self._open = chopper.open
+        self._close = chopper.close
+        self._distance = chopper.distance
+        self._phase = chopper.phase
+        self._name = chopper.name
+        self._open_times = chopper.open_times
+        self._close_times = chopper.close_times
+        super().__init__()
+
+    @property
+    def frequency(self) -> sc.Variable:
+        return self._frequency
+
+    @property
+    def open(self) -> sc.Variable:
+        return self._open
+
```

Comment:
Values of these can of course all be changed, since `sc.Variable` is mutable. But maybe that is fine?